worlds/archipidle/__init__.py

```diff
@@ -38,24 +40,24 @@ class ArchipIDLEWorld(World):
     location_name_to_id = {}
     start_id = 9000
     for i in range(1, 101):
-        location_name_to_id[f"IDLE for at least {int(i / 2)} minutes {30 if (i % 2) > 0 else 0} seconds"] = start_id
+        location_name_to_id[f"IDLE for at least {int(i / 2)} minutes {30 if (i % 2) else 0} seconds"] = start_id
         start_id += 1
 
     def generate_basic(self):
         item_table_copy = list(item_table)
         self.multiworld.random.shuffle(item_table_copy)
+        self.prog_items = set()
 
-        item_pool = []
         for i in range(100):
             item = ArchipIDLEItem(
                 item_table_copy[i],
                 ItemClassification.progression if i < 20 else ItemClassification.filler,
                 self.item_name_to_id[item_table_copy[i]],
                 self.player
             )
-            item_pool.append(item)
-
-        self.multiworld.itempool += item_pool
+            if i < 20:
+                self.prog_items.add(item_table_copy[i])
+            self.multiworld.itempool.append(item)
```

Comment:
i don't see why you would change that (as part of this PR, without providing numbers to back up why your code is better)? building the local list first and appending in one sweep is a valid way to do it. if the world maintainer has "idle commits" lying around such changes might result in needles merge conflicts for them